src/main/java/com/jagrosh/jmusicbot/commands/music/skipSegCmd.java

```diff
@@ -0,0 +1,177 @@
+        Matcher matcher = compiledPattern.matcher(youtubeVideoURL);
+        if (matcher.find()) {
+            return matcher.group();
+        } else {
+            return "fail"; //probably not a youtube video
+        }
+    }
+
+    /**
+     * Gets the current time stamp of the song in milliseconds
+     * 
+     * @param handler audio handler object to extract the current time
+     * @return returns time in milliseconds
+     */
+    private long getCurrentTimeStamp(AudioHandler handler) {
+        return handler.getPlayer().getPlayingTrack().getPosition();
+    }
+
+    /**
+     * Takes string response from api, and parses the JSON to extract any skippable
+     * segments. Compares start and end of segment to current time.
+     * If current time is within the bounds of a segment, return the time stamp of
+     * the end of that segment so it canbee skipped to.
+     * 
+     * @param jsonString Raw string of API response
+     * @param curTime    current time of track
+     * @return returns end of current skippable segment in milliseconds. If no
+     *         segment is currently playing, return -1
+     */
+
+    private long parseMusicJSON(String jsonString, long curTime) {
+        JSONObject obj = new JSONObject(jsonString);
+        JSONArray segmentArr = obj.getJSONArray("segments");
+        for (int i = 0; i < segmentArr.length(); i++) {
+            Float segStart = segmentArr.getJSONObject(i).getFloat("startTime");
+            Float segEnd = segmentArr.getJSONObject(i).getFloat("endTime");
+            if (curTime >= segStart && curTime <= segEnd) {
+                return (long) (segEnd * 1000);
+            }
+        }
+        return -1; //no skippable segment currently playing
+
+    }
+
+    /**
+     * connect to sponsorblock api and attempt to make a GET call to the
+     * searchSegments option. Attempts to get the end of whatever segment
+     * the player is currently in so the player can skip ahead.
+     * 
+     * @param videoID string of the video id for the song that is currently playing
+     *                off youtube
+     * @param handler audio handler to get current time
+     * @param event   message event to send error responses
+     * @return either returns the end of the current segment in milliseconds, -2 if
+     *         no segments were found, or -1 if segments were found but
+     *         the track is not currently within a segment.
+     * @throws UnsupportedEncodingException
+     * @throws IOException
+     */
+    public long connectToAPI(String videoID, AudioHandler handler, CommandEvent event)
+            throws UnsupportedEncodingException, IOException {
+        HttpURLConnection con = null;
+        long curTime = getCurrentTimeStamp(handler) / 1000; // convert to seconds
+        boolean musicSegment = false;
+
+        try {
+            String nonMusicURLString = "https://sponsor.ajay.app/api/searchSegments?videoID=";
```

Comment:
Should use the skipSegments endpoint rather than searchSegments ([docs](https://wiki.sponsor.ajay.app/w/API_Docs#GET_.2Fapi.2FsearchSegments)). Using the [hash prefix version of skipSegments](https://wiki.sponsor.ajay.app/w/API_Docs#GET_.2Fapi.2FskipSegments.2F:sha256HashPrefix) would be better for user privacy.

Also, `category` should be supplied as a query parameter with the value of `music_offtopic` (or `categories` if multiply types are to be skipped) to ensure you're getting non-music segments back, as the default is to only retrieve `sponsor` segments. ([docs](https://wiki.sponsor.ajay.app/w/Types))